client/packages/electron/src/electron.ts

```diff
@@ -70,19 +65,23 @@ class Scanner {
   }
 
   scanDevices(window: BrowserWindow) {
-    const devices: BarcodeScanner[] = [];
     // if a scanner is already connected, we'll need to close it in order to open it
     if (this.device) {
       this.device?.close();
     }
 
-    HID.devices().forEach(device => {
-      devices.push({ ...device, connected: false });
+    this.scanDevicesAsync(window);
```

Comment:
changing to async isn't necessary! however, the detect scanner function worked first time and usually requires it to be run twice, so I kept it in. there's still a strange lag when you use the detect however. that's what I was hoping to resolve with the async.